### DISCUSSION TITLE
webpack5 module federation with splitchunks meet some problems

### DISCUSSION THREAD

**Discussion by lanmch** [webpack/webpack#16571] [Q&A]
Recently, I try to use w5 Module Federation in my project, but after i use this MF plugin（without any module expose or share or use remotes pkg), and change the entry from sync to async. Like,
```javascript
// before
/* index.js */
import React from 'react';
import ReactDOM from 'react-dom';
import App from './app';
ReactDOM.render(<App />, document.getElementById('root'));

// after
/* index.js */
import('./bootstrap');
/* bootstrap.js */
import React from 'react';
import ReactDOM from 'react-dom';
import App from './app';
ReactDOM.render(<App />, document.getElementById('root'));
```

the splitchunk config,  like `splitchunk.chunks` or `cachegroup.{item}.chunks`. these config may be 'all' or 'async', but when the entry change to async, the config might look like not work as before. the bundle size become larger. I think may be some package just build to async chunk before, but build to async chunk and sync chunk now.

**Comment by alexander-akait** [comment]
Can you cliarfy what is the problem? Module duplication? The module federation requires some changes for better code splitting in some cases, it is hard to be align with any posssible things. Maybe you can provide configuration (or exampke when bundled size is larger)?
